dts/bindings/misc/zephyr,swdp-gpio.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2019, Phytec Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: >
+    This is a representation of the Serial Wire Debug Port interface
+    implementation by GPIO bit-banging.
+
+compatible: "zephyr,swdp-gpio"
+
+include: base.yaml
+
+properties:
+    clk-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO pin used for SWCLK output
+
+    dout-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO pin used for SWDIO output
+
+    din-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO pin used for SWDIO input
+
+    dnoe-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO pin used for SWDIO NOE output
+
+    noe-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO pin used for SWD NOE output
```

Comment:
Can't these be optional. There are many hardware designs that do not need to use input/output buffers